distributed/comm/ucx.py

```diff
@@ -0,0 +1,366 @@
+"""
+:ref:`UCX`_ based communications for distributed.
+
+See :ref:`communcations` for more.
+
+.. _UCX: https://github.com/openucx/ucx
+"""
+import asyncio
+import itertools
+import logging
+import struct
+
+from .addressing import parse_host_port, unparse_host_port
+from .core import Comm, Connector, Listener, CommClosedError
+from .registry import Backend, backends
+from .utils import ensure_concrete_host, to_frames, from_frames
+from ..utils import ensure_ip, get_ip, get_ipv6, nbytes
+
+import ucp
+
+import os
+
+os.environ.setdefault("UCX_RNDV_SCHEME", "put_zcopy")
```

Comment:
I haven't followed this discussion closely. Do we plan to allow users to set this in the dask config file?